### ISSUE TITLE
Proposal: Make Icon of NavigationViewItem templatable

### ISSUE THREAD

**Issue by JustinXinLiu** [microsoft/microsoft-ui-xaml#3101]
# Proposal: Make Icon of NavigationViewItem templatable

## Summary

Make `Icon` a type of `object` that can take any element, and add an `IconTemplate` property that allows for full customization. The max width of the `Icon` element should respect the value of `CompactPaneLength`.

## Rationale

* The current `Icon` property only accepts elements of type `IconElement`, which limits its usage.

## Scope

| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to display any type of elements they want in the Icon area. | Must |

**Comment by Felix-Dev** [comment]
One specific example where this would be helpful is developers wanting to show a *PersonPicture* element as the leftmost element of a NavigationViewItem (see [here](https://github.com/microsoft/microsoft-ui-xaml/issues/2541) and [here](https://docs.microsoft.com/en-us/answers/questions/64597/uwp-personpicture-gets-cut-when-implemented-on-a-n.html?childToView=64821#answer-64821) for examples).

Due ue to the current restrictions of the `NavigationViewItem.Icon` API developers have to use the `NavigationViewItem.Content` API instead. If they want to show a content string next to the PdersonPicture, they will have to do all the work for that:
```xml
<NavigationViewItem>
    <NavigationViewItem.Content>
        <StackPanel Orientation="Horizontal">
            <PersonPicture Width="20" Height="20" />
            <TextBlock Text="Account" Margin="10,0,0,0" />
        </StackPanel>       
    </NavigationViewItem.Content>
</NavigationViewItem>
```
![image](https://user-images.githubusercontent.com/1398851/90192126-93208f80-ddc2-11ea-956d-d67a878ffd71.png)

It would be better if developers could just do this instead:
```xml
<NavigationViewItem Content="Account">
    <NavigationViewItem.Icon>
        <PersonPicture Width="20" Height="20" />
    </NavigationViewItem.Icon>
</NavigationViewItem>
```

**Comment by MarcoEnxuto** [comment]
Exactly. I read this post earlier, i found this would be the closest to my question #3116

**Comment by ad1Dima** [comment]
I think this is better than #1488

**Comment by Javier118** [comment]
Hello:
you could add Svg support for icons in NavigationViewItem in this winUi3 version.

**Comment by ad1Dima** [comment]
> you could add Svg support for icons in NavigationViewItem in this winUi3 version.
@Javier118  In UWP ImageSource supports SVG files. If WinUI3 version of ImageSource doesn't support it, you'd better to make new proposal for this.
